app/models/user.rb

```diff
@@ -416,6 +416,22 @@ def is_protected_user?
     has_role?(:protected_user)
   end
 
+  def password_resets_limit_reached?
```

Comment:
I think the ticket mentions something about pulling this out into a [`concern`](https://api.rubyonrails.org/classes/ActiveSupport/Concern.html); I could see that being useful if `Admin` also needs this behaviour at some point, but I don't know if that's planned at the moment